### ISSUE TITLE
auto switch mode bug

### ISSUE THREAD

**Issue by GoogleCodeExporter** [FelisCatus/switchy#54]
```
What steps will reproduce the problem?
1.make a profile name Puff_PAC, choose the "automatic configuration", url, 
http://autoproxy2pac.appspot.com/pac/puff
2.make a profile name Puff_ALL, choose the "manual configuration", url, 
127.0.0.1:1984
3.Enable switch rules, default rule: all urls = Puff_PAC.
4.Rule: imdb, *.imdb.com* = Puff_ALL

What is the expected output?

they are both working fine, when I choose one profile to let it work.
Puff_PAC use puff when the pac told him so. (won't use Puff for sits 
haven't been GFWed. like, http://www.ip-adress.com/)
Puff_ALL use puff for every sites.
but the pac file miss some sites blocked by GFW, I need some custom rules.
I use the automatic mode, settings is just like what I said above.

so, it should be working with Puff_PAC, when I visit *.imdb.com*, it uses 
Puff_ALL.


What do you see instead?

But with the automatic mode, I seems using Puff_ALL for every sites.
I test it with http://www.ip-adress.com/, the automatic mode, it shows the 
proxy server's location, with Puff_PAC works alone, it shows my ip address.

So, there is something wrong with Switchy's automatic mode.

Switchy version: Extension Info: v1.4.2.1
Operating system: Browser Info: 5.0 (Windows; U; Windows NT 6.1; en-US) 
AppleWebKit/532.8 (KHTML, like Gecko) Chrome/4.0.295.0 Safari/532.8


Error Log (Switchy Options > press 'Error Log' button): 

[23:02:09] [info] - Extension Info: v1.4.2.1
[23:02:09] [info] - Browser Info: 5.0 (Windows; U; Windows NT 6.1; en-US) 
AppleWebKit/532.8 (KHTML, like Gecko) Chrome/4.0.295.0 Safari/532.8
[23:02:09] [success] - Plugin loaded successfully..
[23:02:09] [success] - Plugin working properly..
[23:02:09] [success] - 'localStorage' supported..
[23:02:09] [success] - Wrote to local storage successfully..


```

Original issue reported on code.google.com by `fangzhen...@gmail.com` on 21 Jan 2010 at 5:22
- Merged into: #34
